Modules/Core/Sources/Core/Stream/Stream.swift

```diff
@@ -1,28 +1,44 @@
 public enum StreamError : Error {
-    case closedStream(buffer: Buffer)
-    case timeout(buffer: Buffer)
+    case closedStream
+    case timeout
 }
 
 public protocol InputStream {
     var closed: Bool { get }
+    func open(deadline: Double) throws
     func close()
     
-    func read(into: UnsafeMutableBufferPointer<UInt8>, deadline: Double) throws -> Int
-    func read(upTo: Int, deadline: Double) throws -> Buffer
+    func read(into readBuffer: UnsafeMutableBufferPointer<Byte>, deadline: Double) throws -> UnsafeBufferPointer<Byte>
+    func read(upTo byteCount: Int, deadline: Double) throws -> Buffer
 }
 
+
 extension InputStream {
-    public func read(into: UnsafeMutableBufferPointer<UInt8>) throws -> Int {
-        return try read(into: into, deadline: .never)
+    public func read(upTo byteCount: Int, deadline: Double) throws -> Buffer {
+        var bytes = [Byte](repeating: 0, count: byteCount)
+
+        let readBuffer = try bytes.withUnsafeMutableBufferPointer {
```

Comment:
Maybe it's just me but this looks like it could possible cause memory access issues. You pass a pointer into the read function which returns a sub pointer.

The pointer from [UInt8] is only valid for the scope of the closure so unless you're allocating a new buffer inside the read and doing a memcpy this *could* break.